feat: add sentry and redux-sentry-middleware

nathanstitt opened the pull request:
And a small wrapper to configure and set the env for it.

I'm curious of the best way to get set the DEPLOY_ENV, I suspect that reading `global` like I am doing is the wrong way, but not sure how to do better.

nathanstitt committed:
add sentry and redux-sentry-middleware

And a small wrapper to configure and set the env for it

nathanstitt commented on src/config.js:
```diff
@@ -15,6 +15,7 @@
 let config = {
   APP_ENV: process.env.REACT_APP_ENV,
+  DEPLOYED_ENV: global && global.location ? global.location.hostname.split('.')[0] : 'server',
   ACCOUNTS_URL: process.env.ACCOUNTS_URL || 'https://accounts.openstax.org',
```
this seems bad to me.  Do you know of a better alternative?

nathanstitt replied:
well, webpack will rewrite `global` to be `window` for the web.  Using `global` allows it to also be set for non-dom environments. 

We use the first part of the URL in tutor just so it's shorter but maybe that won't work well with with REX targeted domains.  I'll use the entire hostname for now

TomWoodward replied:
`typeof(window) === undefined` returns false in node, thats the pattern we follow throughout the code, doesn't rely on using `global`

nathanstitt replied:
Hm, so using `window` causes specs to fail even though it's behind the typeof check.
```
$ REACT_APP_ENV=test jest --coverage --testPathPattern="(\.|/)spec\.tsx?"
ReferenceError: window is not defined
    at Object.<anonymous> (/Users/nas/code/ox/rex-web/src/config.js:18:59)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
```

We could abandon the attempt at the DEPLOYED_ENV, Sentry will log whatever the url was so we'll still have the info.

TomWoodward replied:
this is because i led you astray, `typeof` returns a string, so its `typeof(window) === 'undefined'`

nathanstitt replied:
oh derp, I knew that.  That's what I get for blindly copy & pasting code 😁

nathanstitt committed:
order imports

TomWoodward committed:
Search toolbar (#191)

* Initial styles and js for search input
 desktop and mobile

* wip react state

* tweaking styles on search toolbar

* add polyfill selector for focus-within

* drying toolbar

* fixes for toc

* Reset toc button styles

* :camera:

* unit tests for toolbar search

* add label to toggle button

TomWoodward commented on src/helpers/Sentry.ts:
```diff
@@ -0,0 +1,32 @@
+
+  initializeWithMiddleware() {
+    Sentry.init({
+      dist: config.CODE_VERSION,
```
release_id is more descriptive, that captures code version and books that are built

TomWoodward committed:
fix submitting search with enter key (#196)

nathanstitt commented on src/app/index.tsx:
```diff
@@ -91,6 +92,8 @@ export default (options: AppOptions) => {
 
   const middleware: Middleware[] = [
     navigation.createMiddleware(routes, history),
+    Sentry.initializeWithMiddleware(),
```
I don't think so, but maybe you see something I don't?  It runs fine in the specs at least.

TomWoodward replied:
can you try running `REACT_APP_ENV=test yarn serve:static` to see if pre-rendering works?

nathanstitt replied:
guessing you meant `start-static`, since that's the only command that has `static` in it.  I ran:
`REACT_APP_ENV=test yarn run start:static`
without any issues and then loaded the site from http://localhost:62873/

TomWoodward replied:
yea sorry thats right, ok cool.

nathanstitt committed:
use complete hostname for DEPLOYED_ENV

nathanstitt committed:
use RELEASE_ID for dist

nathanstitt committed:
set DEPLOYED_ENV

TomWoodward committed:
feat: add search hook and integrate into history state (#185)

* add search hook and persist search state in sessionStorage

* test search hook

* remove persistState

* sync search query into the history state

* tests pass

* wip search navigation state

* better approach, rely on contentlink to forward search state so you don't have to replace all willy nilly

* taking notes to future me

* update notes to future me

* carry over search state when navigating to book content

* clear search state when navigating outside the book

* don't use history directly unnecessarily

* don't clear search if there is nothing in it

* cleanup

* test coverage for search hooks

* work on contentlink specs

* test search state forwarding in ContentLink

* fix cc issues

* reduce cc of hook

* Revert "reduce cc of hook"

This reverts commit cea6ff2a0e6c9bdb9ad963f86d2e2622f50735b5.

TomWoodward committed:
release.json > environment.json (#197)

nathanstitt committed:
test for window === undefined vs using global

nathanstitt committed:
typeof returns string, use that

TomWoodward committed:
add set -e to top of search build script (#194)

TomWoodward reviewed (approved):
this is looking fine, just be aware that its gonna send stuff to sentry from local dev environments, we may want to throw a check somewhere for `REACT_APP_ENV=production` which will be true in every environment thats not local dev or heroku

berkalpyakici committed:
fix: add code style (#198)

nathanstitt committed:
Add IS_PRODUCTION and only log to sentry when it's set to true

berkalpyakici committed:
fix: h3 font size (#192)

berkalpyakici committed:
fix: problem spacing (#200)

* fix: add code style

* fix: add margin to problem numbers

* remove other pr

pyup-bot committed:
Scheduled monthly dependency update for July (#204)

* Update certifi from 2019.3.9 to 2019.6.16

* Update chromedriver-binary from 75.0.3770.8.0 to 76.0.3809.25.0

* Update identify from 1.4.3 to 1.4.5

* Update importlib-metadata from 0.17 to 0.18

* Update more-itertools from 7.0.0 to 7.1.0

* Update pre-commit from 1.16.1 to 1.17.0

* Update pytest from 4.5.0 to 5.0.0

* Update pytest-html from 1.20.0 to 1.21.1

* Update pyyaml from 5.1 to 5.1.1

* Update virtualenv from 16.6.0 to 16.6.1

Malar-Natarajan committed:
Test attribution status in new page (#172)

* tests attribution status in next and previous pages

* verifying next link locators

* next and previous works but with sleep

* playing around toc

* playing around toc

* playing with toc

* toc chapter click works

* page click and chapter click works

* test works but with sleep

* test works good without sleep

* test works good without sleep

* test works

* some minor changes not affecting the test

* updated based on test_attribution review comment

* removed unused attributes

* next element click with title verification works as helper function but still code duplication in base.py

* Next link helper function kind of works

* helper functions created for TOC links

* removed a redundant offscreen click method in pages/base.py
Still have the original method as commented out. Can revert if my changes are wrong

* finishing C476304
removed TOC navigation from this test. It will be covered later

* Deleting content_item.py 

This file was needed for the TOC changes. Since the TOC part is removed from this PR, I am deleting this file

* updated review comments

* refactored click_and_wait to use a reference object and renamed the method

* added offscreen_click back to the region

* refactored content page to use updated click and check reference object on base page

* update docstring on click_and_check helper method

* Revert "update docstring on click_and_check helper method"

This reverts commit adf82be7cb08cd6a55f024bcebaf61de0d3a3986.

* updated the title to be in base.py

TomWoodward committed:
add lcov for codecov (#211)

nathanstitt committed:
only call initialize on Sentry when it's enabled

TomWoodward commented:
looks good now, but code coverage took a hit. after running `yarn test:unit` locally you can open the coverage report with `yarn coverage-report`

nathanstitt commented:
> looks good now, but code coverage took a hit.

Something really weird is going on with coverage.  I ran it as you'd suggested and noticed that the report was pointing out stuff that was no longer in the codebase.  I then thought that maybe coverage wasn't re-generating so I removed all the report files (`rm -r coverage/*`), and they're not re-created after test:unit

Thinking this might be something I'd introduced, I switched to `master` but see the same thing happen there.

It seems like jest dies right as it's generating the coverage. I see multiple:
```
Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
    at process.target.send (internal/child_process.js:627:16)
    at reportSuccess (/Users/nas/code/ox/rex-web/node_modules/jest-worker/build/child.js:59:11)
Emitted 'error' event at:
    at process.nextTick (internal/child_process.js:631:35)
    at process._tickCallback (internal/process/next_tick.js:61:11)
events.js:167
      throw er; // Unhandled 'error' event
```

TomWoodward commented:
sometimes jest workers get into a weird state and i have to do `killall node` to get new ones

nathanstitt committed:
test warn and error methods

TomWoodward commented:
@nathanstitt i'm still showing some things that need coverage:

```
Statements   : 99.78% ( 1360/1363 )
Branches     : 99.63% ( 534/536 )
Functions    : 100% ( 456/456 )
Lines        : 99.76% ( 1252/1255 )
```

TomWoodward committed:
extend domVisitor to support multiple audit logics (#203)

TomWoodward committed:
feat: add logic for generating search results data structure (#207)

* add functions for differentiating types of tree nodes

* add utils for organizing search results by page

* real data structure for search results

* fix prerender

* adding tests for search utils

* :100:s

nathanstitt committed:
test isEnabled conditional branch

TomWoodward committed:
update search type (#214)

TomWoodward commented:
@nathanstitt getting closer...

```
=============================== Coverage summary ===============================
Statements   : 99.93% ( 1362/1363 )
Branches     : 99.81% ( 535/536 )
Functions    : 100% ( 456/456 )
Lines        : 99.92% ( 1254/1255 )
================================================================================
```

andymcheng committed:
fix: font size during landscape orientation (#201)

* fix: add code style

* fix for font size being too large in landscape mode on iphone safari browser

* [wip] fix: landscape mobile browser font size

* removed unncessary webkit

* migrated change to src/index.css

moved text-size-adjust fix from index.less file to src/index.css file as it applys to the whole app

nathanstitt committed:
revert whitespace only changes

nathanstitt committed:
test sentry is initialized and middleware added

nathanstitt committed:
fix lint error

nathanstitt committed:
add sentry/browser

nathanstitt committed:
Merge branch 'master' into log-errors

nathanstitt committed:
lint

TomWoodward commented on src/config.development.js:
```diff
@@ -1,6 +1,7 @@
 module.exports = {
   RELEASE_ID: 'development',
   CODE_VERSION: 'development',
+  DEPLOYED_ENV: 'development',
```
the heroku environments run in development but it'd be nice to be able to differentiate them in the logs. will we be supplying `hostname` in any other metadata? can we supply `APP_ENV` as metadata separately and not override `DEPLOYED_ENV` here?

nathanstitt committed:
Merge branch 'master' into log-errors

nathanstitt committed:
remove imports
